packages/cohere/src/cohere-chat-language-model.ts

```diff
@@ -47,7 +110,58 @@ export class CohereChatLanguageModel implements LanguageModelV2 {
     return this.config.provider;
   }
 
-  private getArgs({
+  private async extractDocuments(prompt: LanguageModelV2Prompt): Promise<
+    Array<{
+      data: { text: string; title?: string };
+    }>
+  > {
+    const documents: Array<{ data: { text: string; title?: string } }> = [];
+
+    for (const message of prompt) {
+      if (message.role === 'user') {
+        for (const part of message.content) {
+          if (part.type === 'file') {
```

Comment:
wonder if it makes sense to limit the supported media types and then go by media type in a switch instead